### PR TITLE
Add terradoc

### PR DESCRIPTION
**Pull request by kassianh**
Recreated README with terradoc, checked documentation for errors or inconsistencies.

### CODE REVIEW

File: README.tfdoc.hcl [hunk 1]
```diff
@@ -0,0 +1,585 @@
...
+          Specifies whether resources in the module will be created.
+        END
+      }
+
+      variable "module_tags" {
+        type        = map(string)
+        default     = {}
+        description = <<-END
+          A map of tags that will be applied to all created resources that accept tags. Tags defined with 'module_tags' can be
+            overwritten by resource-specific tags.
```
Comment (soerenmartius) [inline README.tfdoc.hcl:117]:
```suggestion
          A map of tags that will be applied to all created resources that accept tags. Tags defined with 'module_tags' can be overwritten by resource-specific tags.
```

File: README.tfdoc.hcl [hunk 2]
```diff
@@ -0,0 +1,585 @@
...
+        END
+      }
+
+      variable "module_depends_on" {
+        type        = list(any)
+        description = <<-END
+          A list of dependencies.
+            Any object can be assigned to this list to define a hidden external dependency.
```
Comment (soerenmartius) [inline README.tfdoc.hcl:125]:
```suggestion
          A list of dependencies. Any object can be assigned to this list to define a hidden external dependency.
```

File: README.tfdoc.hcl [hunk 3]
```diff
@@ -0,0 +1,585 @@
...
+          description    = <<-END
+            A Set of objects representing Principals in an IAM policy document.
+            *(only required if `assume_role_policy` is not set)*
+          END
+          readme_example = <<-END
+            assume_role_principals = [
+              { type        = "Service"
+                identifiers = [ "ec2.amazonaws.com" ]
+              }
+            ]
```
Comment (soerenmartius) [inline README.tfdoc.hcl:197]:
```suggestion
            assume_role_principals = [
              {
                type        = "Service"
                identifiers = [ "ec2.amazonaws.com" ]
              }
            ]
```

File: README.tfdoc.hcl [hunk 4]
```diff
@@ -0,0 +1,585 @@
...
+          description    = <<-END
+            A Set of objects representing Conditions in an IAM policy document.
+            *(only evaluated when `assume_role_principals` is used)*
+          END
+          readme_example = <<-END
+            assume_role_conditions = [
+              { test     = "Bool"
+                variable = "aws:MultiFactorAuthPresent"
+                values   = [ "true" ]
+              }
```
Comment (soerenmartius) [inline README.tfdoc.hcl:213]:
```suggestion
            assume_role_conditions = [
              {
                test     = "Bool"
                variable = "aws:MultiFactorAuthPresent"
                values   = [ "true" ]
              }
```

**Review by soerenmartius — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -102,23 +107,27 @@ See [variables.tf] and [examples/] for details and use-cases.
...
+- [**`name_prefix`**](#var-name_prefix): *(Optional `string`)*<a name="var-name_prefix"></a>
 
   If omitted, Terraform will assign a random, unique name.
-  Invalid characters will be replaced with dashes.
-  Creates a unique name beginning with the specified prefix.
-  Conflicts with name.
+Invalid characters will be replaced with dashes.
+Creates a unique name beginning with the specified prefix.
+Conflicts with name.
+Forces new resource.
```
Comment (mariux) [inline README.md:123]: @thiesen can we render this with indent? the result will be the same so no real issue.. but makes the markdown more readable..

File: README.md [hunk 6]
```diff
@@ -177,77 +194,87 @@ See [variables.tf] and [examples/] for details and use-cases.
...
   List of IAM policy statements to attach to the role as an inline policy.
 
+  Example:
+
   ```hcl
   policy_statements = [
     {
       sid = "FullS3Access"
-
+
```
Comment (mariux) [inline README.md:269]: @thiesen looks like we add indent to empty lines which should not happen....

**Review by mariux — APPROVED** [review]
lgtm

**Pull request merged**